### PR TITLE
C# 9.0

### PR DESCRIPTION
**Pull request by amazingalek**
* New "SDK style" projects
* C# 9.0
* .Net Framework 4.7.2 -> 4.8
* Sexy C# 8/9 code

### CODE REVIEW

File: src/OWML.Logging/ConsoleUtils.cs [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
 		{
 			if (string.IsNullOrEmpty(line))
 			{
 				return;
 			}
 
-			Console.ForegroundColor = MessageTypeColors.ContainsKey(type)
-				? MessageTypeColors[type]
-				: ConsoleColor.Gray;
+			Console.ForegroundColor = type switch
```
Comment (misternebula) [inline src/OWML.Logging/ConsoleUtils.cs:15]: Is it just me being used to older c# versions or is this less readable? :p

Comment (misternebula) [reply src/OWML.Logging/ConsoleUtils.cs:15]: especially with the discard _

File: src/OWML.ModHelper.Assets/ModAssets.cs [hunk 2]
```diff
@@ -136,10 +133,7 @@ private IEnumerator LoadTexture(TextureAsset modAsset, string imagePath)
 				yield return www;
 				www.LoadImageIntoTexture(texture);
 			}
-			if (texture == null)
-			{
-				_console.WriteLine("Error - Texture is null.", MessageType.Error);
-			}
```
Comment (misternebula) [inline src/OWML.ModHelper.Assets/ModAssets.cs:142]: Removing error logs...?

File: src/OWML.ModHelper.Events/Patches.cs [hunk 3]
```diff
@@ -39,7 +39,7 @@ internal static class Patches
 		public static void AfterDestroy(MonoBehaviour __instance) =>
 			OnEvent?.Invoke(__instance, Common.Events.AfterDestroy);
 
-		public static IEnumerable<CodeInstruction> EmptyMethod(IEnumerable<CodeInstruction> instructions) => 
+		public static IEnumerable<CodeInstruction> EmptyMethod(IEnumerable<CodeInstruction> _) =>
```
Comment (misternebula) [inline src/OWML.ModHelper.Events/Patches.cs:42]: you sure harmony doesn't need it to be called "instructions" ?

File: src/OWML.ModHelper.Input/ModInputLibrary.cs [hunk 4]
```diff
@@ -13,22 +13,15 @@ public static class ModInputLibrary
...
-					return "correct combination";
-				default:
-					return "unknown";
-			}
-		}
+				RegistrationCode.InvalidCombination => "contains invalid keys",
+				RegistrationCode.CombinationTooLong => "contains too many keys",
+				RegistrationCode.CombinationTaken => "similar combination already registered",
+				RegistrationCode.AllNormal => "correct combination",
+				_ => "unknown"
```
Comment (misternebula) [inline src/OWML.ModHelper.Input/ModInputLibrary.cs:23]: Could just replace with Enum.GetName. less lines, but doesn't explain it as well i guess...

File: src/OWML.ModHelper.Interaction/ModInteraction.cs [hunk 5]
```diff
@@ -74,15 +74,12 @@ public IList<IModBehaviour> GetDependencies(string uniqueName)
...
-					return null;
-				case TInterface inter:
-					return inter;
-				default:
-					return _proxyFactory.CreateProxy<TInterface>(api, _manifest.UniqueName, uniqueName);
-			}
+				null => null,
+				TInterface inter => inter,
+				_ => _proxyFactory.CreateProxy<TInterface>(api, _manifest.UniqueName, uniqueName)
+			};
```
Comment (misternebula) [inline src/OWML.ModHelper.Interaction/ModInteraction.cs:82]: yeah, just not a fan of how these new return switches look :/

File: src/OWML.Logging/ConsoleUtils.cs (hunk 1 above)
Comment (amazingalek) [reply src/OWML.Logging/ConsoleUtils.cs:15]: Maybe it's just you? :P I think it's great :D

File: src/OWML.ModHelper.Assets/ModAssets.cs (hunk 2 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Assets/ModAssets.cs:142]: VS or ReSharper claims texture cannot be null here and I believe it.

File: src/OWML.ModHelper.Events/Patches.cs (hunk 3 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Events/Patches.cs:42]: Yep

File: src/OWML.ModHelper.Input/ModInputLibrary.cs (hunk 4 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Input/ModInputLibrary.cs:23]: I agree but I prefer not to *change* Tai's stuff, as I don't really understand it. I like cleaning it up, but not actually changing the functionality :P

File: src/OWML.ModHelper.Interaction/ModInteraction.cs (hunk 5 above)
Comment (amazingalek) [reply src/OWML.ModHelper.Interaction/ModInteraction.cs:82]: I always hated the old switch syntax and fell in love with switch expressions right away. They're the main reason I upgraded the projects and C# versions :P

**Review by misternebula — APPROVED** [review]

**Pull request merged**